src/AWS.Distro.OpenTelemetry.AutoInstrumentation/LambdaWrapper.cs

```diff
@@ -0,0 +1,188 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#if !NETFRAMEWORK
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using Amazon.Lambda.Core;
+using Newtonsoft.Json.Linq;
+using OpenTelemetry.Instrumentation.AWSLambda;
+using OpenTelemetry.Trace;
+
+// Assembly attribute to enable the Lambda function's JSON input to be converted into a .NET class.
+[assembly: LambdaSerializer(typeof(Amazon.Lambda.Serialization.Json.JsonSerializer))]
+
+namespace AWS.Distro.OpenTelemetry.AutoInstrumentation;
+
+/// <summary>
+/// LambdaWrapper class. This class is used by the instrument.sh file to wrap around Lambda Functions
+/// and forceFlush spans at the end of the lambda function execution
+/// </summary>
+public class LambdaWrapper
+{
+    private static readonly TracerProvider? TracerProvider;
+
+    static LambdaWrapper()
+    {
+        Type? instrumentationType = Type.GetType("OpenTelemetry.AutoInstrumentation.Instrumentation, OpenTelemetry.AutoInstrumentation");
+
+        if (instrumentationType == null)
+        {
+            Console.WriteLine("instrumentationType Type was not found");
+            return;
+        }
+
+        FieldInfo? tracerProviderField = instrumentationType.GetField("_tracerProvider", BindingFlags.Static | BindingFlags.NonPublic);
+
+        if (tracerProviderField == null)
+        {
+            Console.WriteLine("Field '_tracerProvider' not found in Instrumentation class.");
+        }
+
+        object? tracerProviderValue = tracerProviderField?.GetValue(null);
+
+        if (tracerProviderValue != null)
+        {
+            TracerProvider = (TracerProvider)tracerProviderValue;
+        }
+    }
+
+    /// <summary>
+    /// Wrapper function handler for that handles all types of events.
+    /// </summary>
+    /// <param name="input">Input as JObject which will be converted to the correct object type during runtime</param>
+    /// <param name="context">ILambdaContext lambda context</param>
+    /// <returns>returns an object that will be contain the same structure as the original handler output type</returns>
+    public Task<object?> TracingFunctionHandler(JObject input, ILambdaContext context)
+    => AWSLambdaWrapper.TraceAsync(TracerProvider, this.FunctionHandler, input, context);
```

Comment:
Not needed since if if the original handler isn't async, we wrap it as well with the same function. This avoids having the customer to set different handlers for async vs sync.